Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -186,6 +184,289 @@ degenerate_faces(const TriangleMesh& tm, OutputIterator out)
   return degenerate_faces(tm, get(vertex_point, tm), Kernel(), out);
 }
 
+/// \ingroup PMP_repairing_grp
+/// checks whether a vertex is non-manifold.
+///
+/// @tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph`
+///
+/// @param v the vertex to check whether is degenerate
+/// @param tm the triangulated surface mesh upon evaluation
+///
+/// \return true if the vertrex is non-manifold
+template <typename TriangleMesh>
+bool is_non_manifold_vertex(typename boost::graph_traits<TriangleMesh>::vertex_descriptor v,
+                            const TriangleMesh& tm)
+{
+  CGAL_assertion(CGAL::is_triangle_mesh(tm));
+
+  typedef boost::graph_traits<TriangleMesh> GT;
+  typedef typename GT::halfedge_descriptor halfedge_descriptor;
+
+  boost::unordered_set<halfedge_descriptor> halfedges_handled;
+  halfedge_descriptor start = halfedge(v, tm);
+  halfedge_descriptor h=start;
+  do{
+    halfedges_handled.insert(h);
+    h=opposite(next(h, tm), tm);
+  }while(h != start);
+
+  BOOST_FOREACH(halfedge_descriptor h, halfedges_around_target(v, tm))
+  {
```

Comment:
this loop is exactly the one above.
The first loop needs to be on all halfedges of the mesh and you should add all halfedges having `v` as target in `halfedges_handled`.